packages/network-controller/src/NetworkController.ts

```diff
@@ -161,6 +170,12 @@ export class NetworkController extends BaseControllerV2<
     );
   }
 
+  destroy() {
+    // Prevent out-of-band calls to the provider
```

Comment:
The "out-of-band" part seems inaccurate; this prevents _any_ provider operations, out-of-band or not.

Reply:
I just removed this comment. It's obvious that when you stop the provider, it should prevent further calls. [7201172](https://github.com/MetaMask/core/pull/1039/commits/7201172879c99d2ae2887cf351736c39a2b31105)